crawl-ref/source/invent.cc

```diff
@@ -1153,7 +1153,9 @@ bool item_is_selected(const item_def &i, int selector)
                 && !item_is_equipped(i);
 
     case OSEL_CURSABLE:
-        return item_is_cursable(i);
+        // the extra check is to prevent cursing a wielded non-weapon
```

Comment:
Do we actually need to prevent that? It seems like a deeply goofy challenge condition that some people might enjoy. Does it break stuff?

Reply:
We don't need to prevent that here, only because if you try to wield something curable that isn't a weapon slot item you can no longer wield those things! The remaining wieldable items are either weapons or staves, or aren't cursable (evokers and consumables).